packages/hash/api/src/model/knowledge/link.model.ts

```diff
@@ -0,0 +1,160 @@
+import { GraphApi } from "@hashintel/hash-graph-client";
+
+import { EntityModel, LinkModel, LinkTypeModel } from "../index";
+
+export type LinkModelConstructorParams = {
+  accountId: string;
+  sourceEntity: EntityModel;
+  linkTypeModel: LinkTypeModel;
+  targetEntity: EntityModel;
+};
+
+export type LinkModelCreateParams = {
+  accountId: string;
+  sourceEntity: EntityModel;
+  linkTypeModel: LinkTypeModel;
+  targetEntity: EntityModel;
+};
+
+/**
+ * @class {@link LinkModel}
+ */
+export default class {
+  accountId: string;
+
+  sourceEntity: EntityModel;
+  linkTypeModel: LinkTypeModel;
+  targetEntity: EntityModel;
+
+  constructor({
+    accountId,
+    sourceEntity,
+    linkTypeModel,
+    targetEntity,
+  }: LinkModelConstructorParams) {
+    this.accountId = accountId;
+
+    this.sourceEntity = sourceEntity;
+    this.linkTypeModel = linkTypeModel;
+    this.targetEntity = targetEntity;
+  }
+
+  /**
+   * Create a link between a source and a target entity using a specific link
+   * type.
+   *
+   * @param params.accountId the accountId of the account creating the link
+   * @param params.sourceEntity the source entity of the link
+   * @param params.linkTypeModel the Link Type of the link
+   * @param params.targetEntity the target entity of the link
+   */
+  static async create(
```

Comment:
I think this model is the most confusing to me, why is it possible to instantiate a model without the link being created, and if it is already created then why do we have a `create` method? I think my confusion is just stemming from me not understanding the point of the model classes properly in the first place

Reply:
This method is static, so it wouldn't be relevant to instances. I do agree that we shouldn't allow instantiating a model without the link being created, perhaps we should aim for making the constructor `private` or something similar? (I think that might mess with some of the inheritance stuff being done in the model class, but I can make it `private` for now)

Reply:
Apologies I had failed to notice it was static 🙃 

I don't have a strong opinion on your suggestion, if you think it'd be a good idea we can do so, or we can experiment with it at a later date

Reply:
I've added this task: https://app.asana.com/0/1202805690238892/1202890446280565/f to explore forcing proper usage of the model classes